config/config_mgmt.py

```diff
@@ -213,6 +222,69 @@ def writeConfigDB(self, jDiff):
 
         return
 
+    def add_module(self, yang_module_str):
+        """
+        Validate and add new YANG module to the system.
+
+        Parameters:
+            yang_module_str (str): YANG module in string representation.
+
+        Returns:
+            None
+        """
+
+        module_name = self.get_module_name(yang_module_str)
+        module_path = os.path.join(YANG_DIR, '{}.yang'.format(module_name))
+        if os.path.exists(module_path):
+            raise Exception('{} already exists'.format(module_name))
```

Comment:
caller of add_module should be able to handle it gracefully.

Reply:
The caller can catch the exception and handle it gracefully.